File: member-c/admin-web/src/api/notification.ts
import request from './request'
import type { PageResult } from './user'

export interface NotificationItem {
  id: number
  title: string
  content: string
  notificationType: string
  bizType: string
  bizId: number
  readStatus: string
  createdAt: string
  readAt: string
}

export interface UnreadCount {
  noticeCount: number
  todoCount: number
  totalCount: number
}

export function getNotifications(params: {
  page: number
  pageSize: number
  type?: string
  readStatus?: string
}) {
  return request.get<any, PageResult<NotificationItem>>('/notifications', { params })
}

export function getUnreadCount() {
  return request.get<any, UnreadCount>('/notifications/unread-count')
}

export function markAsRead(id: number) {
  return request.put(`/notifications/${id}/read`)
}

export function createNotification(data: {
  receiverId: number
  title: string
  content: string
  notificationType?: string
  bizType?: string
  bizId?: number
}) {
  return request.post('/notifications', data)
}

File: information_school_platform-main/member-c/src/main/java/com/ischool/isp/entity/Notification.java
package com.ischool.isp.entity;

import com.baomidou.mybatisplus.annotation.IdType;
import com.baomidou.mybatisplus.annotation.TableId;
import com.baomidou.mybatisplus.annotation.TableName;
import lombok.Data;

import java.time.LocalDateTime;

@Data
@TableName("notification")
public class Notification {

    @TableId(type = IdType.AUTO)
    private Long id;

    private Long receiverId;

    private String title;

    private String content;

    private String notificationType;

    private String bizType;

    private Long bizId;

    private String readStatus;

    private LocalDateTime createdAt;

    private LocalDateTime readAt;
}
